src/main/java/com/google/sps/servlets/SentimentServlet.java

```diff
@@ -52,40 +63,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     } catch (GoogleJsonResponseException e) { // some other problem with requesting comment data
       response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, COMMENTS_FAILED_ERROR);
     }
-    String comments = String.join(". ", commentsList);
 
     String captions = captionService.getCaptionFromId(videoId);
 
-    if (comments.isEmpty() && captions.isEmpty()) {
-      VideoAnalysis videoAnalysis = new VideoAnalysis.Builder()
-          .setScore(null)
-          .setScoreAvailable(false)
-          .build();
-      String json = new Gson().toJson(videoAnalysis);
-
+    if (commentsList.isEmpty() && captions.isEmpty()) {
+      String json = createResponseJson(null, false);
       response.setContentType("application/json;");
       response.getWriter().println(json);
       return;
     }
+    
+    // Creates 26 thread pools (25 for comments, 1 for captions)
+    ExecutorService executor = Executors.newFixedThreadPool(26);
 
-    String text = comments + ". " + captions;
+    List<Future> commentScoreFutures = new ArrayList<>();
+    for (String comment : commentsList) {
+      Future<Float> future = executor.submit(new Callable<Float>() {
+        @Override
+        public Float call() {
+          return calculateSentimentScore(comment);
+        }
+      });
+      commentScoreFutures.add(future);
+    }
 
-    Float score = null;
+    Future<Float> captionFuture = executor.submit(new Callable<Float>() {
+      @Override
+      public Float call() {
+        return calculateSentimentScore(captions);
+      }
+    });
+    
+    Float commentsScore = 0f;
+    Float captionsScore = 0f;
 
     try {
-      score = calculateSentimentScore(text);
-    } catch (ApiException e) {
+      for (Future<Float> future : commentScoreFutures) {
```

Comment:
you can probably simplify this using [Stream::average](https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html#average--)

Reply:
ack, decided not to do it because handling exceptions in a stream required a lot just to calculate an average :/